src/scene.js

```diff
@@ -24,20 +24,22 @@ var vec3 = glMatrix.vec3;
 if (Utils.isMainThread) {
     // On main thread only (skip in web worker)
     Utils.requestAnimationFramePolyfill();
+
 }
+
 Scene.tile_scale = 4096; // coordinates are locally scaled to the range [0, tile_scale]
 Geo.setTileScale(Scene.tile_scale);
 GLBuilders.setTileScale(Scene.tile_scale);
 GLProgram.defines.TILE_SCALE = Scene.tile_scale;
 
 // Load scene definition: pass an object directly, or a URL as string to load remotely
-export default function Scene(source, config_source, options) {
-
+export default function Scene(config_source, options) {
     options = options || {};
     subscribeMixin(this);
 
     this.initialized = false;
-    this.tile_source = source;
+    this.sources = new Map();
```

Comment:
Haven't checked yet, but there is a good chance this is going to break iOS / Safari. If so we'll need to avoid (unfortunately).